src/lib/UI/Value/ValueFactory.php

```diff
@@ -199,9 +244,11 @@ public function createLocation(Location $location): UIValue\Content\Location
     {
         $translations = $location->getContent()->getVersionInfo()->languageCodes;
         $target = (new Target\Version())->deleteTranslations($translations);
+        $limit = $this->configResolver->getParameter('subtree_operations.query_subtree.limit');
+        $useLimit = $limit > 0;
 
         return new UIValue\Content\Location($location, [
-            'childCount' => $this->locationService->getLocationChildCount($location),
+            'childCount' => $this->locationService->getLocationChildCount($location, $useLimit ? $limit + 1 : null),
```

Comment:
We should be able to establish at this point that current `Location` child count exceeds the limit, maybe?

I'd prefer to set this value to `null` indicating that we aren't able to determine the amount if possible.